src/Microsoft.DotNet.Build.Tasks/VersionTools/UpdateToRemoteDependencies.cs

```diff
@@ -0,0 +1,128 @@
+using Microsoft.DotNet.VersionTools.Automation;
+using Microsoft.DotNet.VersionTools.Automation.GitHubApi;
+using Microsoft.DotNet.VersionTools.Dependencies.BuildOutput;
+using Microsoft.DotNet.VersionTools.Util;
+using System;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks.VersionTools
+{
+    public class UpdateToRemoteDependencies : BaseDependenciesTask
+    {
+        public string CurrentRefXmlPath { get; set; }
+
+        /// <summary>
+        /// If provided, GitHub authentication info is used to fetch the remote dotnet/versions
+        /// commit. The anonymous user rate limit is small, and this can be used to use an account's
+        /// quota instead. The "...AndSubmitPullRequest" subclass also uses this to create the PR.
+        /// </summary>
+        public string GitHubAuthToken { get; set; }
+        public string GitHubUser { get; set; }
+        public string GitHubEmail { get; set; }
+
+        protected override void TraceListenedExecute()
+        {
+            using (GitHubClient client = CreateClient(allowAnonymous: true))
+            {
+                DependencyUpdateResults updateResults = UpdateToRemote(client);
+
+                Log.LogMessage(
+                    MessageImportance.Low,
+                    $"Suggested commit message: '{updateResults.GetSuggestedCommitMessage()}'");
+            }
+        }
+
+        protected GitHubClient CreateClient(bool allowAnonymous)
+        {
+            GitHubAuth gitHubAuth = null;
+            if (string.IsNullOrEmpty(GitHubAuthToken))
+            {
+                if (allowAnonymous)
+                {
+                    Log.LogMessage(
+                        MessageImportance.Low,
+                        $"No value for '{nameof(GitHubAuthToken)}'. " +
+                        "Accessing GitHub API anonymously.");
+                }
+                else
+                {
+                    throw new ArgumentException(
+                        $"Required argument '{nameof(GitHubAuthToken)}' is null or empty.");
+                }
+            }
+            else
+            {
+                gitHubAuth = new GitHubAuth(GitHubAuthToken, GitHubUser, GitHubEmail);
+            }
+
+            return new GitHubClient(gitHubAuth);
+        }
+
+        protected DependencyUpdateResults UpdateToRemote(GitHubClient client)
+        {
+            // Use the commit hash of the remote dotnet/versions repo master branch.
+            string versionsCommitHash = client
+                .GetReferenceAsync(new GitHubProject("versions", "dotnet"), "heads/master")
+                .Result.Object.Sha;
+
+            DependencyUpdateResults updateResults = DependencyUpdateUtils.Update(
+                CreateUpdaters().ToArray(),
+                CreateDependencyInfos(true, versionsCommitHash).ToArray());
+
+            // Update CurrentRef and CurrentBranch for each applicable build info used.
```

Comment:
Seems like you aren't using CurrentBranch any longer, so perhaps the comment is stale.